tests/unit/mathematics/SGLinalg_unittest.cc

```diff
@@ -60,6 +60,21 @@ TEST(SGLinalg, CPUBackend_dot)
 	EXPECT_NEAR(result, 20.0, 1E-15);
 }
 
+TEST(SGLinalg, SGVector_CPUBackend_dot)
```

Comment:
Well, when you change the behavior of this `convert_vector` thing (return gpu type if there is any, otherwise go on with cpu type), then this test should work even outside `HAVE_VIENNACL` guard.